pyomo/contrib/incidence_analysis/incidence.py

```diff
@@ -0,0 +1,106 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+"""Functionality for identifying variables that are "incident on" expressions
+"""
+import enum
+from pyomo.core.expr.visitor import identify_variables
+from pyomo.repn import generate_standard_repn
+from pyomo.common.backports import nullcontext
+from pyomo.util.subsystems import TemporarySubsystemManager
+from pyomo.contrib.incidence_analysis.config import IncidenceMethod, IncidenceConfig
+
+
+#
+# Handlers for different methods of generating the incidence graph
+#
+def _get_incident_via_identify_variables(expr, include_fixed):
+    # Note that identify_variables will not identify the same variable
+    # more than once.
+    return list(identify_variables(expr, include_fixed=include_fixed))
+
+
+def _get_incident_via_standard_repn(expr, include_fixed, linear_only):
+    if include_fixed:
+        to_unfix = [
+            var for var in identify_variables(expr, include_fixed=True) if var.fixed
+        ]
+        context = TemporarySubsystemManager(to_unfix=to_unfix)
+    else:
+        context = nullcontext()
+
+    with context:
+        repn = generate_standard_repn(expr, compute_values=False)
```

Comment:
If you never explicitly need the quadratic terms, why not run `generate_standard_repn(expr, compute_values=False, quadratic=False)`?  Doing that would allow you to remove all the quadratic handling below.

Reply:
Good point. Updated to use `quadratic=False`